packages/components/package.json

```diff
@@ -23,12 +23,11 @@
     "tailwind.config.js"
   ],
   "scripts": {
-    "build": "yarn run build:clean && yarn run build:js && yarn run build:styles && yarn run build:types && yarn run build:finish",
+    "build": "yarn run build:clean && yarn run build:js && yarn run build:styles && yarn run build:finish",
     "build:clean": "rm -rf lib/",
     "build:finish": "rm lib/**/*.spec.* lib/**/*.stories.*",
-    "build:js": "babel src -d lib --extensions '.js,.jsx,.ts,.tsx'",
+    "build:js": "tsc",
```

Comment:
I first looked at using esbuild for compilation, which is totally doable. But we would still need to use `tsc` to get the type definitions, so we may as well also use `tsc` to compile to js.

Reply:
few things I want to check:
1. input / output directories are now fully controlled by `tsconfig.json`?
2. are declaration files still being output into `lib/`?
3. should we consider using the `--build` or `--project` flag?

Reply:
Good questions.

> input / output directories are now fully controlled by tsconfig.json?

That's right, the in and out directories are not fully controlled by tsconfig.json (which was already configured to do so. Probably just to match the babel config).

> are declaration files still being output into lib/?

Yep, declaration files are still being output. And we're now getting sourcemaps output, too. This technically makes the package larger, but they're pretty small and may be useful for debugging in LP.

> should we consider using the --build or --project flag?

We don't need `--project`, since it's for specifying a specific tsconfig.json, but it'll find the one in the package's root directory no problem.

We also don't need `--build` as currently written. That's for use with multiple typescript projects which we're not making use of right now. Could think about it if we added more packages.

https://github.com/chanzuckerberg/frontend-libs does use `--build`, and there are some nice and annoying things about it.